src/main/java/net/dv8tion/jda/api/interactions/components/buttons/Button.java

```
@@ -230,7 +240,26 @@ default Button withUrl(@Nonnull String url)
     {
         Checks.notEmpty(url, "URL");
         Checks.notLonger(url, URL_MAX_LENGTH, "URL");
-        return new ButtonImpl(null, getLabel(), ButtonStyle.LINK, url, isDisabled(), getEmoji());
+        return new ButtonImpl(null, getLabel(), ButtonStyle.LINK, url, null, isDisabled(), getEmoji());
+    }
+
+    /**
+     * Returns a copy of this button with the provided SKU.
+     *
+     * @param  sku
+     *         The SKU to use
+     *
+     * @throws IllegalArgumentException
+     *         If the provided {@code sku} is null
+     *
+     * @return New button with the changed url
+     */
+    @Nonnull
+    @CheckReturnValue
+    default Button withSku(@Nonnull SkuSnowflake sku)
+    {
+        Checks.notNull(sku, "SKU");
+        return new ButtonImpl(null, getLabel(), ButtonStyle.PREMIUM, null, sku, isDisabled(), null);
```

This should throw an exception if the current style is not premium.